src/which.js

```diff
@@ -45,7 +40,17 @@ function _which(options, cmd) {
   var where = null;
 
   // No relative/absolute paths provided?
-  if (cmd.search(/\//) === -1) {
+  if (cmd.indexOf('/') === -1) {
+    // Assume that there are no extensions to append to queries (this is the
+    // case for unix)
+    var pathExtArray = [''];
+    if (common.platform === 'win') {
+      // In case the PATHEXT variable is somehow not set (e.g.
+      // child_process.spawn with an empty environment), use the XP default.
+      var pathExtEnv = process.env.PATHEXT || XP_DEFAULT_PATHEXT;
+      pathExtArray = splitPath(pathExtEnv.toUpperCase());
+    }
+
     // Search for command in PATH
     pathArray.forEach(function (dir) {
```

Comment:
Yeah, I agree. Index-based for loops kind of suck in JS, but it's probably still better than this exit-early-if-we-found-it approach. I will adjust.